shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/core/NetworkClient.kt

```diff
@@ -51,4 +42,28 @@ class NetworkClient {
             emit(data)
         }
     }
+
+    inline fun <reified T, reified M> fetchWithMeta(builder: HttpRequestBuilder) : Flow<Pair<T, M>> {
+        return flow {
+            val body = client.request(builder).bodyAsText()
+            val data = JsonApi(json).decodeWithMetaFromJsonApiString<T, M>(body)
+            emit(data)
+        }
+    }
+
+    private fun clientConfig(): HttpClientConfig<*>.() -> Unit {
+        return {
+            install(Logging)
+            install(ContentNegotiation) {
+                json(json)
+            }
+            install(Auth) {
+                bearer(bearer())
+            }
+        }
+    }
```

Comment:
I prefer to open this fun instead. Different network client has different config.

```suggestion
    open fun clientConfig(): HttpClientConfig<*>.() -> Unit {
        return {
            install(Logging)
            install(ContentNegotiation) {
                json(json)
            }
        }
    }
```

and in `TokenizedNetworkClient`
```kotlin
    open fun clientConfig(): HttpClientConfig<*>.() -> Unit {
        return {
            install(Logging)
            install(ContentNegotiation) {
                json(json)
            }
			install(Auth) {
                bearer(bearer())
            }
        }
    }

	private fun bearer(): BearerAuthConfig.() -> Unit {
        return {
            loadTokens {
                localDataSource.getToken().singleOrNull()?.run {
                    BearerTokens(this.accessToken, this.refreshToken)
                }
            }
            sendWithoutRequest { request ->
                val builder = HttpRequestBuilder()
                builder.url("${BuildKonfig.BASE_URL}")
                request.url.host != builder.url.host
            }
        }
    }
```